### PR TITLE
docs: add documentation for Artifactory provider

### PR DESCRIPTION
**Pull request by jhosteny**

### CODE REVIEW

File: provider/artifactory/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+| SIDECRED_ARTIFACTORY_PROVIDER_HOSTNAME | String | No | N/A | Artifactory endpoint (e.g., `https://my-org.jfrog.io/my-org/`) |
+| SIDECRED_ARTIFACTORY_PROVIDER_USERNAME | String | No | N/A | REST API authentication username |
+| SIDECRED_ARTIFACTORY_PROVIDER_PASSWORD | String | Yes | N/A | REST API authentication password |
+| SIDECRED_ARTIFACTORY_PROVIDER_ACCESS_TOKEN | String | Yes | N/A | REST API access token |
+| SIDECRED_ARTIFACTORY_PROVIDER_API_KEY | String | Yes | N/A | REST API key |
+| SIDECRED_ARTIFACTORY_PROVIDER_SESSION_DURATION | Duration | Yes | `1h` | Default duration for generate tokens (`<= `h`) |
+
+The fields marked as not optional assume that the provider is enabled.
+
+### Resource
```
[the remark this answers is not in the document]
Comment (jhosteny) [reply provider/artifactory/README.md]: Agreed, that is an oversight on my part

**Comment by jhosteny** [comment]
Hi @itsdalmo, thanks for the review!

I agree with what you proposed, and will work on the godoc changes. I probably won't get to it until this weekend or early next week though.

**Comment by itsdalmo** [comment]
Thanks @jhosteny! I'm back from vacation as of yesterday, so I'm hoping to start writing some documentation and terraform over and/or after the weekend 👍

**Comment by jhosteny** [comment]
@itsdalmo sorry for the delay on this.

Would you like the list of supported stores, providers and backends in the top level README to remain there, or be moved to the main package godocs as well? If left, I can add links to the appropriate godocs or local READMEs.

**Pull request merged**